File: examples/typescript/aws-cdk/templates/s3-stack.ts
import { Stack } from '@awslv/aws-cdk-organizer';
import { StackGroup } from '@awslv/aws-cdk-organizer';
import { StackProps } from 'aws-cdk-lib';
import { Bucket } from 'aws-cdk-lib/aws-s3';
import { Construct } from 'constructs';

export type S3StackProps = StackProps & {
  bucketName: string;
};

export class S3Stack extends Stack<S3StackProps> {
  public bucket: Bucket;

  constructor(
    stackGroup: StackGroup,
    scope: Construct,
    id: string,
    props: S3StackProps
  ) {
    super(stackGroup, scope, id, props);

    this.bucket = new Bucket(this, 'Bucket', {
      bucketName: this.getBucketName(props.bucketName)
    });
  }
}

File: examples/python/aws-cdk/templates/s3_stack.py
"""S3 Bucket Stack."""

from dataclasses import dataclass
from typing import TYPE_CHECKING

from aws_cdk import CfnOutput
from aws_cdk.aws_s3 import Bucket

if TYPE_CHECKING:
    from constructs import Construct
    from cdk_organizer.aws.stack_group import StackGroup

from cdk_organizer.aws.stack import Stack


@dataclass
class S3BucketStackParameters:
    base_bucket_name: str


class S3BucketStack(Stack):
    def __init__(
        self,
        scope: "Construct",
        id: str,
        stack_group: "StackGroup",
        parameters: S3BucketStackParameters,
        **kwargs
    ) -> None:
        super().__init__(scope, id, stack_group, **kwargs)

        self.bucket = Bucket(
            self,
            'S3Bucket',
            bucket_name=self.get_bucket_name(parameters.base_bucket_name)
        )

        CfnOutput(self, 'BucketName', value=self.bucket.bucket_name)
        CfnOutput(self, 'BucketArn', value=self.bucket.bucket_arn)
